addon/components/uni-input.js

```diff
@@ -0,0 +1,106 @@
+import Ember from 'ember';
+import layout from '../templates/components/uni-input';
+
+const { Component, computed, isEmpty } = Ember;
+const SUCCESS_CLASS = 'uni-input--success';
+const ERROR_CLASS = 'uni-input--error';
+const WARNING_CLASS = 'uni-input--warning';
+const INPUT_TYPE_NUMBER = 'number';
+const INPUT_TYPE_EMAIL = 'email';
+
+export default Component.extend({
+  layout,
+
+  classNames: ['uni-input-tooltip-container'],
+  defaultClass: ['uni-input uni-input--bordered uni-input--small'],
+  customClass: '',
+  value: null,
+  type: 'text',
+  name: '',
+  placeholder: '',
+  maxLength: null,
+  tooltipMessage: null,
+  showTooltip: false,
+  useDefaultErrorValidation: true,
+  useDefaultSuccessValidation: false,
+  isRequired: false,
+  showSuccess: false,
+  showError: false,
+  showWarning: false,
+  onChange() {},
+
+  isValid: computed('value', function() {
+    switch (this.get('type')) {
+      case INPUT_TYPE_EMAIL:
+        return this._isValidEmail();
+      case INPUT_TYPE_NUMBER:
+        return this.get('isRequired') ? !isNaN(this.get('value')) : true;
+      default: return !(this.get('isRequired') && isEmpty(this.get('value')));
+    }
+  }),
+
+  successClass: computed('showSuccess', function() {
+    if (this.get('useDefaultSuccessValidation')) {
+      return this.get('isValid') ? SUCCESS_CLASS : '';
```

Comment:
if you use type number, it only accepts numbers.
also, you can check isValid since its always computing and I'm passing it on the onChange function.